### DISCUSSION TITLE
cannot find module './attachDirectiveResolvers

### DISCUSSION THREAD

**Discussion by sanshigongA** [requarks/wiki#7378] [Error / Bug Report] [answered]
![image](https://github.com/user-attachments/assets/a284053d-b4f2-4080-831c-cb101744c056)

Windows_NT 6.1.7601【Windows server 2008 R2】
PostgreSQL v11.2
nodejs v12.22.12
npm    v6.14.16
wiki.js  v2.5.304

**Comment by NGPixel** [comment]
Did you run any npm command?

**Reply by sanshigongA** [reply]
I just execute Command "node server"

**Reply by sanshigongA** [reply]
and open to the setup page in my browser，then error

**Reply by sanshigongA** [reply]
> Did you run any npm command?
**The log is:**
……
10 silly lifecycle wiki@2.5.304~start: Args: [ '/d /s /c', 'node server' ]
11 silly lifecycle wiki@2.5.304~start: Returned: code: 1  signal: null
12 info lifecycle wiki@2.5.304~start: Failed to exec start script
13 verbose stack Error: wiki@2.5.304 start: `node server`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (D:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:314:20)
13 verbose stack     at ChildProcess.<anonymous> (D:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:314:20)
13 verbose stack     at maybeClose (internal/child_process.js:1022:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
14 verbose pkgid wiki@2.5.304
15 verbose cwd D:\wiki-js-windows
16 verbose Windows_NT 6.1.7601
17 verbose argv "D:\\Program Files\\nodejs\\node.exe" "D:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "start"
18 verbose node v12.22.12
19 verbose npm  v6.14.16
20 error code ELIFECYCLE
21 error errno 1
22 error wiki@2.5.304 start: `node server`
22 error Exit status 1
23 error Failed at the wiki@2.5.304 start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]

**Reply by sanshigongA** [reply]
> Did you run any npm command?

```
D:\wiki-js-windows>npm list
……
npm ERR! peer dep missing: graphql@^0.10.5 || ^0.11.3 || ^0.12.0 || ^0.13.0 || ^14.0.0, required by graphql-subscriptions@1.1.0
npm ERR! missing: graphql@^14.0.2, required by graphql-rate-limit-directive@1.2.1
npm ERR! error in D:\wiki-js-windows\node_modules\@graphql-tools\graphql-tag-pluck\node_modules\@babel\traverse\node_modules\@babel\types: ENOENT: no
such file or directory, open 'D:\wiki-js-windows\node_modules\@graphql-tools\graphql-tag-pluck\node_modules\@babel\traverse\node_modules\@babel\types\
package.json'
npm ERR! invalid: @babel/types D:\wiki-js-windows\node_modules\@graphql-tools\graphql-tag-pluck\node_modules\@babel\traverse\node_modules\@babel\types

npm ERR! missing: @babel/types@^7.25.6, required by @babel/parser@7.25.6
npm ERR! extraneous: @graphql-tools/merge@8.3.1 D:\wiki-js-windows\node_modules\@graphql-tools\merge\node_modules\@graphql-tools\merge
npm ERR! missing: @graphql-tools/merge@8.3.1, required by @graphql-tools/schema@8.5.1
npm ERR! error in D:\wiki-js-windows\node_modules\@graphql-tools\merge\node_modules\@graphql-tools\schema\node_modules\tslib: ENOENT: no such file or
directory, open 'D:\wiki-js-windows\node_modules\@graphql-tools\merge\node_modules\@graphql-tools\schema\node_modules\tslib\package.json'
npm ERR! invalid: tslib D:\wiki-js-windows\node_modules\@graphql-tools\merge\node_modules\@graphql-tools\schema\node_modules\tslib
npm ERR! missing: tslib@^2.4.0, required by @graphql-tools/utils@8.9.0
npm ERR! error in D:\wiki-js-windows\node_modules\@graphql-tools\relay-operation-optimizer\node_modules\@graphql-tools\utils: ENOENT: no such file or
directory, open 'D:\wiki-js-windows\node_modules\@graphql-tools\relay-operation-optimizer\node_modules\@graphql-tools\utils\package.json'
npm ERR! invalid: @graphql-tools/utils D:\wiki-js-windows\node_modules\@graphql-tools\relay-operation-optimizer\node_modules\@graphql-tools\utils
npm ERR! extraneous: ws@7.4.5 D:\wiki-js-windows\node_modules\isomorphic-ws\node_modules\ws
```

**Reply by BPowell76** [reply, marked as answer]
@sanshigongA are you not able to update your version of Node and, by extension, npm? You're using an outdated and no longer supported version for both. The current node LTS version is 20.18.0. The current npm version along with it is version 10.8.2.

The version of npm you are running is about 6 years old and some of the dependencies for the packages missing are newer than the version of npm you are running.
